eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -2390,6 +2382,23 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
       state
     }
 
+    /**
+     * We don't acknowledge htlc commands immediately, because we send them to the channel as soon as possible, and they
+     * may not yet have been written to the database.
+     * @param cmd fail/fulfill command that has been processed
+     */
+    def acking(channelId: ByteVector32, cmd: HtlcSettlementCommand): FSM.State[fr.acinq.eclair.channel.State, Data] = {
+      log.debug("scheduling acknowledgement of cmd id={}", cmd.id)
+      context.system.scheduler.scheduleOnce(10 seconds)(PendingRelayDb.ackCommand(nodeParams.db.pendingRelay, channelId, cmd))(context.system.dispatcher)
+      state
+    }
+
+    def acking(updates: List[UpdateMessage]): FSM.State[fr.acinq.eclair.channel.State, Data] = {
+      log.debug("scheduling acknowledgement of cmds ids={}", updates.collect { case s: HtlcSettlementMessage => s.id }.mkString(","))
```

Comment:
nit:

```suggestion
      log.debug("scheduling acknowledgement of htlc ids={}", updates.collect { case s: HtlcSettlementMessage => s.id }.mkString(","))
```

Reply:
We really acknowledge commands, don't we?

Reply:
That's true, it makes sense.